src/_micro-api/libraries/signalDecoder/src/signalDecoder.cpp

```diff
@@ -262,10 +175,17 @@ bool SignalDetectorClass::decode(const int * pulse)
 void SignalDetectorClass::compress_pattern()
 {
 	calcHisto();
-	for (uint8_t idx = 0; idx<patternLen; idx++)
+
+	for (uint8_t idx = 0; idx<patternLen-1; idx++)
```

Comment:
Die äußere Schleife geht von 0 - n-1. Die innere von 1 - n. Wobei in der inneren immer nur Elemente nach den Vergleichswert durchsucht werden.
Also in Durchlauf 0 (idx) wird mit 1 - 11 (idx2) verglichen.
Durchlauf 1 (idx) wird mit 2 - 11 (idx2) verglichen. usw.